query/src/main/java/org/infinispan/query/clustered/commandworkers/CQLazyFetcher.java

```diff
@@ -16,8 +19,22 @@
    @Override
    QueryResponse perform(BitSet segments) {
       DocumentExtractor extractor = getQueryBox().get(queryId);
-      Object key = extractKey(extractor, docIndex);
-      Object value = cache.get(key);
+      Object value;
+      try {
+         value = extractValue(extractor, docIndex);
+      } catch (IOException e) {
```

Comment:
Why not catch the IOE /throw SearchException in the extractValue method. extractKey does it like this already